src/librustdoc/html/render.rs

```diff
@@ -1764,6 +1764,18 @@ fn item_module(w: &mut fmt::Formatter, cx: &Context,
     }
 
     indices.sort_by(|&i1, &i2| cmp(&items[i1], &items[i2], i1, i2));
+    indices.dedup_by_key(|i| (items[*i].def_id,
+                              if items[*i].name.as_ref().is_some() {
+                                  Some(full_path(cx, &items[*i]).clone())
+                              } else {
+                                  items[*i].name.clone()
+                              },
```

Comment:
I don't see how this else branch will return anything but None? Also was there any major reason that `is_some` needed the `as_ref` there?